infra/lambda/index.py

```diff
@@ -34,33 +35,31 @@ def handler(event, context):
                 continue
             break
 
-        # Copy the archive tgz from the bucket to the local disk.
-        tmp_archive = tempfile.mktemp(suffix='.tgz')
-        print('| Downloading S3 archive {}/{} to {}...'.format(bucket, archive_key, tmp_archive))
-        s3.meta.client.download_file(bucket, archive_key, tmp_archive)
+        # Download the archive tgz from the bucket and buffer it into memory.
+        # We do this instead of saving it to a temporary file to avoid running into
+        # the Lambda /tmp directory storage limit of 512 MB.
+        print('| Downloading S3 archive {}/{}...'.format(bucket, archive_key))
+        archive_bytes = io.BytesIO(s3.get_object(Bucket=bucket, Key=archive_key)['Body'].read())
```

Comment:
I tried this as is and this did not work - changing as follows fixed it.

```suggestion
        archive_bytes = io.BytesIO(s3.meta.client.get_object(Bucket=bucket, Key=archive_key)['Body'].read())
```